packages/terra-section-header/src/SectionHeader.jsx

```diff
@@ -150,9 +152,12 @@ class SectionHeader extends React.Component {
     // eslint doesn't know about this and so it marks this as a lint error
     /* eslint-disable jsx-a11y/click-events-have-key-events */
     /* eslint-disable jsx-a11y/no-static-element-interactions */
+
+    const buttonAttributes = (onClick) ? { role: 'button', 'aria-expanded': isOpen, 'aria-label': headerText } : undefined;
+
     return (
-      <Element {...attributes} onClick={onClick} className={sectionHeaderClassNames} tabIndex="0">
-        <div role="button" aria-expanded={isOpen} tabIndex="-1" aria-label={headerText} className={cx('arrange-wrapper')}>
+      <Element {...attributes} className={sectionHeaderClassNames} aria-label={!onClick ? headerText : undefined}>
```

Comment:
```suggestion
      <Element {...attributes} className={sectionHeaderClassNames} aria-label={!onClick ? headerText : undefined} tabIndex="0">
```
you would still need a tabIndex to make it accessible for Screen readers and keyboard users. I think header should still be focusable even when it does not contains `onclick()` 
Need to clarify this with A11y team !!

Reply:
Non-interactive ones are not supposed to receive focus as mentioned in guide https://github.com/cerner/terra-core/blob/A11y-MVB/packages/terra-core-docs/src/terra-dev-site/doc/section-header/AccessibilityGuide.4.doc.mdx?plain=1#L76